mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/TaskExecutorState.java

```diff
@@ -100,10 +100,10 @@ boolean onRegistration(TaskExecutorRegistration registration) {
             this.registration = registration;
             this.gateway =
                 rpcService.connect(registration.getTaskExecutorAddress(), TaskExecutorGateway.class)
-                    .whenComplete((gateway, throwable) -> {
-                        if (throwable != null) {
-                            log.error("Failed to connect to the gateway", throwable);
-                        }
+                    .exceptionally((throwable) -> {
+                        log.error("Failed to connect to the gateway onRegistration, reset to unregistered.", throwable);
+                        onDisconnection();
```

Comment:
why disconnect vs reconnect?

Reply:
this is logic inside taskExecutorState and chain the detached future error to another detached future error is doable but I prefer to have the upper actor to make these decisions based on registration/heartbeat events.